### PR TITLE
Avoid conflicts with DS langauge detection prefs

### PR DESCRIPTION
**Pull request by acrollet**
## Description

This PR uses explicit language detection preferences in check-in to avoid conflicts with DS component language detection.

## Original issue(s)
department-of-veterans-affairs/va.gov-team#42559

## Testing done

- manual testing
- added/updated e2e tests

## Acceptance criteria
- [ ] browser language detection works
  - [ ] for our app
  - [ ] for DS components
  - [ ] language picker still works
  - [ ] language picker preference overrides browser detection and is sticky from page to page

### CODE REVIEW

File: src/applications/check-in/utils/i18n/i18n.js [hunk 1]
```diff
@@ -25,8 +34,13 @@ i18n
   .use(initReactI18next)
   .use(LanguageDetector)
   .init({
+    detection: {
+      order: ['localStorage', 'sessionStorage', 'navigator'],
+      lookupLocalStorage: 'checkin-i18nextLng',
```
Comment (acrollet) [inline src/applications/check-in/utils/i18n/i18n.js:39]: or should it be `i18nextLng-checkin` 🤔

Comment (brianseek) [reply src/applications/check-in/utils/i18n/i18n.js:39]: ¯\_(ツ)_/¯ I think this makes sense.

**Review by brianseek — APPROVED** [review]
Works for me!

**Review by brianseek — APPROVED** [review]
Looks good! Nice improvments

**Pull request merged**